worker/src/rebuild.rs

```diff
@@ -128,16 +139,43 @@ pub async fn rebuild(ctx: &Context<'_>, url: &str) -> Result<Rebuild> {
 
     // process result
     let output_path = out_dir.join(&filename);
+
+    info!("Generating signed link");
+
+
+    let signed_link = in_toto_run(
+        &format!("rebuild {}", filename.to_str().unwrap()),
+        None,
+        &[input_path.to_str().unwrap()],
+        &[output_path.to_str().unwrap()],
+        &[],
+        Some(ctx.privkey),
+        Some(&["sha512", "sha256"]),
+    )?;
+
+    info!("Signed link generated");
+
     if !output_path.exists() {
-        info!("Build failed, no output artifact found at {:?}", output_path);
+        info!(
+            "Build failed, no output artifact found at {:?}",
+            output_path
+        );
         Ok(Rebuild::new(BuildStatus::Bad, log))
     } else if compare_files(&input_path, &output_path).await? {
         info!("Files are identical, marking as GOOD");
-        Ok(Rebuild::new(BuildStatus::Good, log))
+        let mut res = Rebuild::new(BuildStatus::Good, log);
+
+        let attestation = serde_json::to_string(&signed_link).context("Failed to serialize attestation")?;
+        res.attestation = Some(attestation);
+
+        Ok(res)
     } else {
         info!("Build successful but artifacts differ");
         let mut res = Rebuild::new(BuildStatus::Bad, log);
 
+        // let attestation = serde_json::to_string(&signed_link).context("Failed to serialize attestation")?;
```

Comment:
This can be deleted I think? Line 168 and 169 seem to be doing this now.

Reply:
Yes! Thank you